kolibri/core/assets/src/views/CoreMenu/__tests__/CoreMenuOption.spec.js

```diff
@@ -0,0 +1,202 @@
+import { render, fireEvent, screen } from '@testing-library/vue';
+import userEvent from '@testing-library/user-event';
+import VueRouter from 'vue-router';
+import CoreMenuOption from '../CoreMenuOption.vue';
+import '@testing-library/jest-dom';
+
+const sampleSubRoutes = [
+  { name: 'subRoute1', label: 'Sub Route 1' },
+  { name: 'subRoute2', label: 'Sub Route 2' },
+  { name: 'subRoute3', label: 'Sub Route 3' },
+];
+const sampleLink = 'https://mockurl.com/';
+const sampleIcon = 'add';
+
+const renderComponent = props => {
+  return render(CoreMenuOption, {
+    props: {
+      subRoutes: [],
+      ...props,
+    },
+    routes: new VueRouter(),
+  });
+};
+
+describe('CoreMenuOption', () => {
+  test('smoke test', () => {
+    renderComponent();
+    expect(screen.getByRole('menuitem')).toBeInTheDocument();
+  });
+
+  describe('subRoutes toggle', () => {
+    describe('when subRoutes are provided', () => {
+      it('should render with the chevronDown icon initially with the subroutes not visible', () => {
+        renderComponent({ subRoutes: sampleSubRoutes });
+
+        expect(screen.getByRole('menuitem')).toBeInTheDocument();
+        expect(screen.getByTestId('icon-chevronDown')).toBeInTheDocument();
+        sampleSubRoutes.forEach(subRoute =>
+          expect(screen.queryByText(subRoute.label)).not.toBeInTheDocument()
+        );
+      });
+
+      it('should open the submenu on clicking and the icons should change accordingly', async () => {
+        renderComponent({ subRoutes: sampleSubRoutes });
+        const menuItem = screen.getByRole('menuitem');
+
+        // Clicking should show the subroutes
+        await fireEvent.click(menuItem);
```

Comment:
Im sorry I havent noticed this before, until now I thought we were using the user-event library :sweat_smile:, but I think Its worth getting used to preferring the userEvent library instead of fireEvent, as it is the [recommended way](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event) for the most use cases, so we can replace all these clicks and keydowns methods of fireEvent with the userEvent.click or userEvent.type methods.

Reply:
Hey! Nice catch. I have updated all the instances of `fireEvent.click` with `userEvent.click` as instructed. The `userEvent` library does not provide a method to simulate a `keyDown` event with respect to a particular element, as we require in the case of `menuitem` and the `Enter` key. I tried using the `.keyboard` and `.type` methods, but they didn't seem to trigger the desired effects.

Reply:
(Just for the reference) It seems that with `userEvent.type` was not working because before typing on the element, it internally clicks on the element to have the focus. And as we have the handler also on the `click` element, it was calling the function twice, so visibleSubMenu=!visibleSubMenu called twice ends in the sub menu not being visible. So its ok to keep the fireEvent here.

Reply:
Ahh got it! Thanks so much for the explanation!